tests/test_buffered_synced_collection.py

```diff
@@ -0,0 +1,421 @@
+
+        assert 'inside_second' in synced_dict
+        assert 'inside_second' in on_disk_dict
+        assert 'inside_second' in synced_dict2
+        assert 'inside_second' in on_disk_dict2
+
+    def test_nested_copied_collection(self, synced_dict):
+        """Test modifying two collections pointing to the same data."""
+        synced_dict2 = BufferedJSONDict(filename=synced_dict._filename)
+
+        assert len(synced_dict) == 0
+        assert len(synced_dict2) == 0
+
+        synced_dict['outside'] = 1
+        with synced_dict.buffered():
+            synced_dict['inside_first'] = 2
+
+            on_disk_dict = self.load(synced_dict)
+            assert synced_dict['inside_first'] == 2
+            assert 'inside_first' not in on_disk_dict
+
+            with buffer_reads_writes():
+                synced_dict['inside_second'] = 3
+                synced_dict2['inside_second'] = 4
+
+                on_disk_dict = self.load(synced_dict)
+                assert synced_dict['inside_second'] == 4
+                assert synced_dict2['inside_second'] == 4
+                assert 'inside_second' not in on_disk_dict
+
+            on_disk_dict = self.load(synced_dict)
+            assert on_disk_dict['inside_second'] == 4
+
+    @pytest.mark.skip("Not currently sure what the expected behavior is.")
+    def test_nested_copied_collection_invalid(self, synced_dict):
+        """Test the behavior of invalid modifications of copied objects."""
+        synced_dict2 = BufferedJSONDict(filename=synced_dict._filename)
+
+        assert len(synced_dict) == 0
+        assert len(synced_dict2) == 0
+
+        synced_dict['outside'] = 1
+        finished = False
+        with pytest.raises(MetadataError):
+            with synced_dict.buffered():
+                synced_dict['inside_first'] = 2
+                # TODO: Currently, modifying synced_dict2 here causes problems.
+                # It is unbuffered, so it directly writes to file. Then, when
+                # entering global buffering in the context below, synced_dict2
+                # sees that synced_dict has already saved data for this file to
+                # the buffer, so it loads that data, which also means that
+                # synced_dict2 becomes associated with the metadata stored when
+                # synced_dict entered buffered mode. As a result, when the
+                # global buffering exits, we see metadata errors because
+                # synced_dict2 lost track of the fact that it saved changes to
+                # filemade prior to entering the global buffer. We _could_ fix
+                # this by changing the behavior of _load_buffer to not load the
+                # data from the cache if it exists, if the object is new to
+                # cached_collections then we would save a new version. However,
+                # I'm not sure that's the correct answer. Is there a true
+                # canonical source of truth in this scenario?
```

Comment:
Wow. I spent some time thinking about this and played with the code on my own machine for a considerable amount of time. I think it's such a weird edge case that I don't want to devote further engineering effort to it. If a user has problems with this sharp edge AND has opinions on how to solve it, I'm happy to consider it with that user's input. I won't personally do such a torturous thing to my buffers and would recommend the same course of action to all users.

We can keep this test and skip it, but I'd rather see it commented out or removed to avoid the appearance of an "optional" skipped test.

Reply:
I'd rather not remove it entirely so that we have a record of a known failure case, but I'm fine punting on a solution since I truly don't know whether there is a suitable choice for the preferred source of truth here. Either using `xfail` or commenting out would be OK with me. @csadorf thoughts?